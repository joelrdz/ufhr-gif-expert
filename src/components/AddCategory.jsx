import { useState } from 'react';

export const AddCategory = () => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const handleSubmit = e => {
    e.preventDefault();
    console.log(inputValue);
  };

  return (
    <form onSubmit={ handleSubmit }>
      <input
        type="text"
        placeholder="Search gifs:"
        value={ inputValue }
        onChange={ handleInputChange }
      />
    </form>
  );
};