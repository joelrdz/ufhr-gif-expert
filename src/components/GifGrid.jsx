import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
  const { gifs, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{ category }</h3>
      { loading && <p>Loading...</p> }
      <div className="card-grid">
        {gifs.map(gif => (
          <GifItem
            key={ gif.id }
            { ...gif }
          />
        ))}
      </div>
    </>
  );
};