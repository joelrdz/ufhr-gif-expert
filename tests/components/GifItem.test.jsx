import { render } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Tests: <GifItem />', () => {
  const title = 'Vegeta';
  const url = 'https://dragon-ball.com/vegeta.jpg';

  test('should match snapshot', () => {
    const { container } = render( <GifItem title={ title } url={ url } /> );
    expect( container ).toMatchSnapshot();
  });
});
