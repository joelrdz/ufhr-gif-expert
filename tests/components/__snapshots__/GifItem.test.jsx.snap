// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests: <GifItem /> should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Vegeta"
      src="https://dragon-ball.com/vegeta.jpg"
    />
    <p>
      Vegeta
    </p>
  </div>
</div>
`;
